tests/kernel/common/src/riscv_atomic.c

```diff
@@ -0,0 +1,98 @@
+/*
```

Comment:
Well, it is basically addition to atomic operations, I even thought of adding it in `/tests/kernel/common/src/atomic.c`
since test environment is the same.
I'll check where it can fit in `tests/arch/riscv`.